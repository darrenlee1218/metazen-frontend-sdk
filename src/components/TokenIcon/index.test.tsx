import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { ComponentProps } from 'react';
import { TokenIcon } from '.';
import { TestProviders } from '@utils/testing/TestProviders';

afterEach(cleanup);

const defaultProps: ComponentProps<typeof TokenIcon> = {
  tokenIconUrl: 'tokenIcon.png',
  networkIconUrl: 'networkIcon.png',
  isNative: false,
};

it('renders network icon if isNative is false', () => {
  render(<TokenIcon {...defaultProps} />, { wrapper: TestProviders });
  expect(screen.getByTestId('network-icon')).toBeInTheDocument();
});

it('does not render network icon if isNative is true', () => {
  const props = { ...defaultProps, isNative: true };
  render(<TokenIcon {...props} />, { wrapper: TestProviders });
  expect(screen.queryByTestId('network-icon')).toBe(null);
});

it('matches snapshot', () => {
  const { container } = render(<TokenIcon {...defaultProps} />, { wrapper: TestProviders });
  expect(container).toMatchSnapshot();
});
