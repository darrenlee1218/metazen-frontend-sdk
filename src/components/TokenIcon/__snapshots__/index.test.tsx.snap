// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <span
    class="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular css-21opj1-MuiAvatar-root"
    >
      <img
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="tokenIcon.png"
      />
    </div>
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightCircular MuiBadge-overlapCircular BaseBadge-badge css-6zrg6j-MuiBadge-badge"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular css-1ycs6pv-MuiAvatar-root"
        data-testid="network-icon"
      >
        <img
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="networkIcon.png"
        />
      </div>
    </span>
  </span>
</div>
`;
